elasticapm/traces.py

```diff
@@ -316,3 +320,20 @@ def set_user_context(username=None, email=None, user_id=None):
     if user_id is not None:
         data['id'] = encoding.keyword_field(user_id)
     set_context(data, 'user')
+
+
+def mark(group_name, name, override=True):
```

Comment:
why not make `mark` an instance method of `transaction`?

Reply:
Generally, the agent user isn't interacting directly with the transaction anywhere, and instead uses these module-level functions (which can also be accessed from the "root" module, so e.g. `elasticapm.mark()`). There's two reasons for this API design:

 * it makes for a nicer API in my opinion. Compare

       elasticapm.mark('my-mark')

   with

       transaction = get_transaction()
       transaction.mark('my-mark')
 * `get_transaction()` is not a public API (as in not documented, there's not really anything we can do to stop people from using it), because at the moment it is a very thin wrapper around a thread local. But once we start supporting async event loops, thread locals will not work anymore, so we will need other ways to get the current transaction. I currently don't know how exactly that will look, so this API gives me some future wiggle room.